internal/cli/internal/watch/watch.go

```diff
@@ -0,0 +1,287 @@
+		tb.Done("Troubleshooting guide at https://fly.io/docs/getting-started/troubleshooting/")
+		return flyerr.ErrAbort
+	}
+
+	return nil
+}
+
+func WatchReleaseCommand(ctx context.Context, id string) error {
+	g, ctx := errgroup.WithContext(ctx)
+	io := iostreams.FromContext(ctx)
+	client := client.FromContext(ctx).API()
+	interactive := io.IsInteractive()
+	appName := app.NameFromContext(ctx)
+
+	s := spinner.New(spinner.CharSets[11], 100*time.Millisecond)
+	s.Writer = os.Stderr
+	s.Prefix = "Running release task..."
+
+	if interactive {
+		s.Start()
+		defer s.Stop()
+	}
+
+	rcUpdates := make(chan api.ReleaseCommand)
+
+	startLogs := func(ctx context.Context, vmid string) {
+		g.Go(func() error {
+			childCtx, cancel := context.WithCancel(ctx)
+			defer cancel()
+
+			opts := &logs.LogOptions{
+				MaxBackoff: time.Second,
+				AppName:    appName,
+				VMID:       vmid,
+			}
+
+			ls, err := logs.NewPollingStream(childCtx, client, opts)
+			if err != nil {
+				return err
+			}
+
+			for entry := range ls.Stream(childCtx, opts) {
+				if interactive {
+					s.Stop()
+				}
+
+				fmt.Fprintln(io.Out, "\t", entry.Message)
+
+				// watch for the shutdown message
+				if entry.Message == "Starting clean up." {
+					cancel()
+				}
+				s.Start()
+			}
+
+			if err = ls.Err(); errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
+				err = nil
+			}
+			return err
+		})
+	}
+
+	g.Go(func() error {
+		var lastValue *api.ReleaseCommand
+		var errorCount int
+		defer close(rcUpdates)
+
+		for {
+			rc, err := client.GetReleaseCommand(ctx, id)
+			if err != nil {
+				errorCount += 1
+				if errorCount < 3 {
+					continue
+				}
+				return err
+			}
+
+			if !reflect.DeepEqual(lastValue, rc) {
+				lastValue = rc
+				rcUpdates <- *rc
+			}
+
+			if !rc.InProgress {
+				break
+			}
+
+			time.Sleep(500 * time.Millisecond)
+		}
+
+		return nil
+	})
+
+	g.Go(func() error {
+		// The logs goroutine will stop itself when it sees a shutdown log message.
+		// If the message never comes (delayed logs, etc) the deploy will hang.
+		// This timeout makes sure they always stop a few seconds after the release task is done.
+		logsCtx, logsCancel := context.WithCancel(ctx)
+		defer time.AfterFunc(3*time.Second, logsCancel)
+
+		for rc := range rcUpdates {
+			if interactive {
+				s.Prefix = fmt.Sprintf("Running release task (%s)...", rc.Status)
+			}
+
+			if rc.InstanceID != nil {
+				startLogs(logsCtx, *rc.InstanceID)
+			}
+
+			if !rc.InProgress && rc.Failed {
+				if rc.Succeeded && interactive {
+					s.FinalMSG = "Running release task...Done\n"
+				} else if rc.Failed {
+					return fmt.Errorf("release command failed, deployment aborted")
+				}
+			}
+		}
+
+		return nil
+	})
+
+	return g.Wait()
+}
+
+func renderLogs(ctx context.Context, alloc *api.AllocationStatus) {
+	out := iostreams.FromContext(ctx).Out
+	cfg := config.FromContext(ctx).JSONOutput
+
+	logPresenter := presenters.LogPresenter{
+		HideAllocID:    true,
```

Comment:
@azazeal do we have something in render to replace this logic?

Reply:
Nope, you should implement something as we don't use `presenters' any more